monai/apps/datasets.py

```diff
@@ -58,7 +61,7 @@ def __init__(
         self,
         root_dir: str,
         section: str,
-        transform: Callable[..., Any] = LoadPNGd("image"),
+        transform: Callable = DEFAULT_LOADPNGD,
```

Comment:
> B008 Do not perform function calls in argument defaults. The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call. If this is intended, assign the function call to a module-level variable and use that variable as a default value.

Is this intended or should I do something like

```python
transform: Optional[Callable] = None

if transform is None:
    transform = LoadPNGd("image")
```

Reply:
the proposed change of the default behaviour will actually break one of the tutorials. looks like we need a solution to support both no transform and default_loading @Nic-Ma

Reply:
Hi @wyli and @benjamin-gorman ,

1. I think we can use `transform: Optional[Callable] = DEFAULT_LOADPNGD` to support both `default transform` and `None` transform.
2. And then I will change the `CacheDataset` and `PersistentDataset` to support no transforms(actually, there is a bug in `PersistentDataset` that it has default value `None` for transform, but can't work with `None`).

What do you think?

Thanks.

Reply:
@Nic-Ma the problem here is that `DEFAULT_LOADPNGD` is a single globally shared instance, may have some surprising side effects. @benjamin-gorman I think we can keep it for now, add `# ignore`if needed. it is an issue and needs another PR to address it

Reply:
@wyli , yes, I think maybe we should not use the global variable for it now.
@benjamin-gorman , I am OK with both solutions:
1. `transform: Optional[Callable] = DEFAULT_LOADPNGD` with clear description for the potential issue.
2. `transform: Optional[Callable] = LoadPNGd("image")`.

Thanks.

Reply:
Reverted the changes and made issue #727